meson.build

```diff
@@ -0,0 +1,405 @@
+            # WinRT/UWP/app build: GDI and LoadLibrary are unavailable,
+            # but DirectWrite is always present
+            deps += cc.find_library('dwrite', required: true)
+            font_providers += ['DirectWrite (WinRT/UWP)']
+        else
+            # Win32/desktop build: GDI is always present;
+            # DirectWrite is optional but can be loaded via LoadLibrary
+            deps += cc.find_library('gdi32', required: true)
+            font_providers += ['DirectWrite and GDI (Win32)']
+        endif
+    endif
+endif
+
+if get_option('directwrite').enabled() and directwrite == false
+    error(
+        'DirectWrite support was requested, but it was not found.',
+    )
+endif
+
+# CoreText
+coretext = false
+if not get_option('coretext').disabled()
+    appservices_dep = dependency(
+        'appleframeworks',
+        modules: ['ApplicationServices', 'CoreFoundation'],
+        required: false,
+    )
+    # this intentionally includes a leading newline
+    coretext_check = '''
+        int main(void) {
+            CTFontDescriptorCopyAttribute(NULL, kCTFontNameAttribute);
+            return 0;
+        }
+    '''
+    appservices_snippet = (
+        '#include <ApplicationServices/ApplicationServices.h>' + coretext_check
+    )
+
+    if appservices_dep.found() and cc.compiles(appservices_snippet)
+        deps += appservices_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+        font_providers += ['CoreText']
+    else
+        coretext_dep = dependency(
+            'appleframeworks',
+            modules: ['CoreText', 'CoreFoundation'],
+            required: false,
+        )
+        coretext_snippet = '#include <CoreText/CoreText.h>' + coretext_check
+        if coretext_dep.found() and cc.compiles(coretext_snippet)
+            deps += coretext_dep
+            conf.set('CONFIG_CORETEXT', 1)
+            coretext = true
+            font_providers += ['CoreText']
+        endif
+    endif
+endif
+
+if get_option('coretext').enabled() and coretext == false
+    error(
+        'CoreText support was requested, but it was not found.',
+    )
+endif
+
+if get_option('require-system-font-provider')
+    if font_providers.length() == 0
+        error(
+            'Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig (Linux, ' +
+            'other) is required. If you really want to compile without a system font ' +
+            'provider, set -Drequire-system-font-provider=false',
+        )
+    endif
+else
+    if font_providers.length() == 0
+        font_providers += 'none'
+    endif
+endif
+
+# ASM
+enable_asm = false
+# used in libass/meson.build
+asm_is_nasm = false
+asm_args = []
+
+# ASM architecture variables
+asm_option = get_option('asm')
+host_cpu_fam = host_machine.cpu_family()
+host_is_x86_32 = host_cpu_fam == 'x86'
+host_is_x86_64 = host_cpu_fam == 'x86_64'
+host_is_x86 = host_is_x86_32 or host_is_x86_64
+host_is_aarch64 = host_cpu_fam == 'aarch64'
```

Comment:
> I disagree with your stance on the host_is_ variables: they are just much more readable standins for logic that was previously repetitiously scattered across many lines, in multiple places

With `host_is_*` we need to add at least one, possible multiple (for subflavours) global variables for each added assembly arch. With just two `cpufamily` and `subflavour` (effectively enum) variables we can cover all current and probably all future assembly arches. Since their names are normalised, there also sin’ŧ any readability difference between checking `cpufamily == 'x86'` or `host_is_x86`.  
For future assembly additions using `CCAS`, a `cpufamily in ['arm64', 'riscv', 'wasm']` check will imho even be more readable and concise than `host_is_arm64 or host_is_riscv or host_is_wasm`.

Reply:
seems good thx